src/components/NavigationAppBar.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import IconButton from '@material-ui/core/IconButton';
+import Typography from '@material-ui/core/Typography';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import MoreIcon from '@material-ui/icons/MoreVert';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+import InfoIcon from '@material-ui/icons/Info';
+import InputIcon from '@material-ui/icons/Input';
+import CategoryIcon from '@material-ui/icons/Category';
+import ShowChartIcon from '@material-ui/icons/ShowChart';
+import PublishIcon from '@material-ui/icons/Publish';
+import { FormattedMessage } from 'react-intl';
+import styled from 'styled-components';
+
+const Root = styled.div`
+  width: 100%;
+`;
+
+const Grow = styled.div`
+  flex-grow: 1;
+`;
+
+const SectionDesktop = styled.div`
+  display: none;
+  /* @TODO define what width mobile is */
+  @media (min-width: 60rem) {
+    display: flex;
+  }
+`;
+
+const SectionMobile = styled.div`
+  display: flex;
+  @media (min-width: 60rem) {
+    display: none;
+  }
+`;
+
+const Title = styled.div`
+  display: none;
+  @media (min-width: 37.5rem) {
+    display: block;
+  }
+`;
+
+class NavigationAppBar extends React.Component {
+  state = {
+    mobileMoreAnchorEl: null
+  };
+
+  handleMobileMenuOpen = event => {
+    this.setState({ mobileMoreAnchorEl: event.currentTarget });
+  };
+
+  handleMobileMenuClose = () => {
+    this.setState({ mobileMoreAnchorEl: null });
+  };
+
+  handleMenuClick = pageIndex => {
+    const { onChangePage } = this.props;
+    this.handleMobileMenuClose();
+    onChangePage(pageIndex);
+  };
+
+  render() {
+    const { mobileMoreAnchorEl } = this.state;
+    const { onChangePage } = this.props;
+    const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);
+
+    const renderMobileMenu = (
+      <Menu
+        anchorEl={mobileMoreAnchorEl}
+        anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'right' }}
+        open={isMobileMenuOpen}
+        onClose={this.handleMobileMenuClose}
+      >
+        <MenuItem onClick={() => this.handleMenuClick(0)}>
```

Comment:
I think there is an opportunity to refactor this MenuItem into an own component, you can decide if you would like to do it now or add a todo and refactor it later.